src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnectionOptions.cs

```diff
@@ -28,7 +28,14 @@ public class HttpConnectionOptions
         public HttpConnectionOptions()
         {
             _headers = new Dictionary<string, string>();
-            _clientCertificates = new X509CertificateCollection();
+
+            try
+            {
+                _clientCertificates = new X509CertificateCollection();
```

Comment:
You could check for Browser specifically:

```C#
if (!OperatingSystem.IsOSPlatform("browser"))
{
   _clientCertificates = new X509CertificateCollection();
}
```

The compat analyzer should complain if there are other unsupported platforms.

/cc @jeffhandley

Reply:
Ah true, that will be a little better than try catch